.pre-commit-config.yaml

```diff
@@ -1,34 +1,23 @@
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.4.0
+    rev: v4.5.0
     hooks:
       - id: check-added-large-files
       - id: check-yaml
       - id: check-merge-conflict
 
-  # Automatic source code formatting
-  - repo: https://github.com/psf/black
-    rev: 23.3.0
-    hooks:
-      - id: black
-        args: [--safe, --quiet]
-        files: \.pyi?
-        types: [file]
-
   - repo: local
     hooks:
       - id: ruff
-        name: Run ruff
-        stages: [commit]
+        name: lint with ruff
         language: system
-        entry: ruff
+        entry: ruff check --force-exclude
         types: [python]
+        require_serial: true
 
-  # Re-enable after https://github.com/DiamondLightSource/dodal/issues/311
-  # - repo: https://github.com/pre-commit/mirrors-mypy
-  #   rev: v1.7.1
-  #   hooks:
-  #     - id: mypy
-  #       files: 'src/.*\.py$'
-  #       additional_dependencies: [types-requests, pydantic]
-  #       args: ["--ignore-missing-imports", "--show-traceback", "--no-strict-optional"]
```

Comment:
Should: If we're going to remove this can we close https://github.com/DiamondLightSource/dodal/issues/311?